multimodal/src/autogluon/multimodal/utils/onnx.py

```diff
@@ -7,99 +7,70 @@
-        The default batch to help trace and export the model.
+    return dynamic_axes
 
-    """
-    default_onnx_path = None
-    if pipeline == FEATURE_EXTRACTION:
-        valid_input = [
-            "hf_text_text_token_ids",
-            "hf_text_text_segment_ids",
-            "hf_text_text_valid_length",
-        ]
-        dynamic_axes = {
-            "hf_text_text_token_ids": {
-                0: "batch_size",
-                1: "sentence_length",
-            },
-            "hf_text_text_valid_length": {
-                0: "batch_size",
-            },
-            "hf_text_text_segment_ids": {
-                0: "batch_size",
-                1: "sentence_length",
-            },
-        }
-        if config:
-            default_onnx_path = config["model"]["hf_text"]["checkpoint_name"].replace("/", "_") + ".onnx"
-        default_batch = {
-            "hf_text_text_token_ids": tensor(
-                [
-                    [101, 1037, 2158, 2003, 2652, 2858, 1012, 102, 0, 0, 0, 0, 0, 0],
-                    [101, 1037, 2158, 2003, 2652, 1037, 2858, 1012, 102, 0, 0, 0, 0, 0],
-                    [101, 1037, 2158, 2003, 2652, 1037, 2858, 1012, 102, 0, 0, 0, 0, 0],
-                    [101, 1037, 2158, 2003, 6276, 2019, 20949, 1012, 102, 0, 0, 0, 0, 0],
-                    [101, 1037, 2158, 2003, 9670, 1012, 102, 0, 0, 0, 0, 0, 0, 0],
-                    [101, 1037, 2158, 2003, 26514, 2330, 1037, 3869, 1012, 102, 0, 0, 0, 0],
-                    [101, 1037, 2158, 2003, 26514, 1037, 20856, 1012, 102, 0, 0, 0, 0, 0],
-                    [101, 1037, 2158, 2003, 2652, 1037, 2858, 1012, 102, 0, 0, 0, 0, 0],
-                ],
-            ),
-            "hf_text_text_valid_length": tensor(
-                [8, 9, 9, 9, 7, 10, 9, 9],
-            ),
-            "hf_text_text_segment_ids": tensor(
-                [
-                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
-                ],
-            ),
-        }
-    elif pipeline == MULTICLASS:
-        valid_input = ["timm_image_image", "timm_image_image_valid_num"]
-        dynamic_axes = {
-            "timm_image_image": {
-                0: "batch_size",
-            },
-            "timm_image_image_valid_num": {},
-        }
-        default_onnx_path = "./model.onnx"
-        default_batch = None
-    else:
-        raise ValueError(f"ONNX export is not supported in current pipeline {pipeline}")
 
-    return valid_input, dynamic_axes, default_onnx_path, default_batch
+class OnnxModule(object):
```

Comment:
Consider adding some docstrings to explain this class. Tensorrt seems only one provider in onnx runtime. But it's unclear how the three providers work. Does onnxruntime check `TensorrtExecutionProvider `, `CUDAExecutionProvider `, and `CPUExecutionProvider ` sequentially to see their availability? Better to use some docstrings to describe the mechanism.

Reply:
> Does onnxruntime check TensorrtExecutionProvider , CUDAExecutionProvider , and CPUExecutionProvider  sequentially to see their availability?

Yes, the execution providers are sequentially checked.

> Better to use some docstrings to describe the mechanism.

Sure.

Reply:
Added docstring for describing the mechanism.